github_workflow.md

```diff
@@ -0,0 +1,153 @@
+# GitHub Milestone and Issue Workflow
+
+This purpose of this document is to describe our development workflow on
+GitHub; in particular, how we use milestones and issues. We recognize that our
+many subprojects may use variations of these ideas. This particular workflow is
+based on the practices from the JupyterLab repo
+(https://github.com/jupyterlab/jupyterlab) and are optional for Jupyter
+subprojects to follow.
+
+## Milestones
+
+We use Github milestones to track issues and pull requests related to particular
+releases of a subproject (https://github.com/jupyterlab/jupyterlab/milestones).
+
+* These milestones should be labeled with the version numbers of the releases
+  (`4.3`, `5.0`, etc.).
+* Subprojects should only have a few numbered milestones. Beyond one or two
+  future releases, we have found that detailed milestones become meaningless.
+* Subprojects should also have a `Backlog` milestone for issues and pull
+  requests that have not been assigned to a particular release milestone. This
+  also includes new feature requests ("wishlist") issues that we haven't
+  committed to implementing. Essentially, `Backlog` means "at some point in the
+  future, maybe."
```

Comment:
We used to use "wishlist" for this, so perhaps a note that "wishlist" has been renamed "backlog", but still means the same thing. The note about "wishlist" here is perhaps a bit confusing, since it doesn't describe what "wishlist" referred to in the past (and present, on most repos).

Maybe a better name would be "unplanned" or "unscheduled"? Backlog to me means that it is an issue in need of triage attention, whereas this is the opposite: it has been considered and explicitly deprioritized.